src/pixelator/pixeldataset/aggregation.py

```diff
@@ -0,0 +1,157 @@
+"""Module for aggregating pixeldatasets.
+
+Copyright (c) 2023 Pixelgen Technologies AB.
+"""
+from __future__ import annotations
+
+import logging
+from typing import (
+    List,
+)
+
+import pandas as pd
+import polars as pl
+from anndata import AnnData
+from anndata import concat as concatenate_anndata
+
+from pixelator.pixeldataset import PixelDataset
+from pixelator.pixeldataset.utils import update_metrics_anndata, _enforce_edgelist_types
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_attr_and_index_by_component(attribute, sample_names, datasets):
+    for name, dataset in zip(sample_names, datasets):
+        attr = getattr(dataset, attribute, None)
+        if attr is not None:
+            attr["sample"] = pd.Categorical([name] * len(attr))
+            attr["component"] = pd.Categorical(
+                attr["component"].astype(str) + "_" + attr["sample"].astype(str)
+            )
+            attr.set_index(["component"])
+            yield attr
+
+
+def _concatenate_edgelists(datasets, sample_names):
+    with pl.StringCache():
+        concatenated = datasets[0].edgelist_lazy
+        concatenated = concatenated.with_columns(
+            sample=pl.lit(sample_names[0], dtype=pl.Categorical)
+        )
+        concatenated = concatenated.collect()
+
+        for idx, subsequent in enumerate(datasets[1:], start=1):
+            concatenated = concatenated.extend(
+                subsequent.edgelist_lazy.collect().with_columns(
+                    sample=pl.lit(sample_names[idx], dtype=pl.Categorical)
+                )
+            )
+
+        concatenated = concatenated.with_columns(
+            component=pl.concat_str(
+                pl.col("component"), pl.col("sample"), separator="_"
+            )
+        )
+
+    return concatenated
+
+
+def simple_aggregate(
+    sample_names: List[str],
+    datasets: List[PixelDataset],
+    ignore_edgelists: bool = False,
+) -> PixelDataset:
+    """Aggregate samples in a simple way (see caveats).
+
+    Aggregating samples in a simplistic fashion. This function should only
+    be used if the dataset you merge have been generated with the same panel.
```

Comment:
Is there anyway we could assert for this in the function? I think it should be possible as we store the panel and could easily check the metadata or the targets.

Reply:
Right now we require that the antibodies are the same, but we probably want to do a more advanced aggregation here. Perhaps you want to add a ticket about that and we can follow up on that?